src/pudl/transform/eia860.py

```diff
@@ -1021,3 +1021,95 @@ def _core_eia860__boiler_stack_flue(
     )
 
     return bsf_assn
+
+
+@asset
+def _core_eia860__cooling_equipment(
+    raw_eia860__cooling_equipment: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transform the EIA 860 cooling equipment table.
+
+    - spot clean year values before converting to dates
+    - standardize water rate units to gallons per minute (2009-2013 used cubic
+      feet per second)
+    - convert kilodollars to normal dollars
+    """
+    ce_df = raw_eia860__cooling_equipment
+
+    # Generic cleaning
+    ce_df = ce_df.pipe(pudl.helpers.fix_eia_na).pipe(pudl.helpers.add_fips_ids)
+
+    # Spot cleaning and date conversion
+    ce_df.loc[
+        ce_df["chlorine_equipment_operating_year"] == 971,
+        "chlorine_equipment_operating_year",
+    ] = 1971
+    ce_df.loc[ce_df["pond_operating_year"] == 0, "pond_operating_year"] = np.nan
+    ce_df.loc[ce_df["tower_operating_year"] == 974, "tower_operating_year"] = 1974
+    ce_df = ce_df.pipe(pudl.helpers.month_year_to_date).pipe(
+        pudl.helpers.convert_to_date
+    )
+    # Convert cubic feet/second to gallons/minute
+    cfs_in_gpm = 448.8311688
+    ce_df = ce_df.fillna(
+        {
+            "tower_water_rate_100pct_gallons_per_minute": ce_df.tower_water_rate_100pct_cubic_feet_per_second
+            * cfs_in_gpm,
+            "intake_rate_100pct_gallons_per_minute": ce_df.intake_rate_100pct_cubic_feet_per_second
+            * cfs_in_gpm,
+        }
+    ).drop(
+        columns=[
+            "tower_water_rate_100pct_cubic_feet_per_second",
+            "intake_rate_100pct_cubic_feet_per_second",
+        ]
+    )
+
+    # Convert thousands of dollars to dollars and remove suffix from column name
+    ce_df.loc[:, ce_df.columns.str.endswith("_thousand_dollars")] *= 1000
+    ce_df.columns = ce_df.columns.str.replace("_thousand_dollars", "")
+
+    return ce_df.pipe(apply_pudl_dtypes, group="eia", strict=True)
+
+
+@asset_check(asset=_core_eia860__cooling_equipment, blocking=True)
+def cooling_equipment_null_cols(cooling_equipment):
+    """The only null cols we expect are tower type 3 and 4."""
```

Comment:
I think this check *should* be fine in fast ETL since we expect tower 3 and tower 4 in recent years. In other cases, I think we should probably get to a point where we're enforcing that the assets have the same schemas regardless of what years they were run for.

Reply:
I more mean if there's a field like `annual_rate` that only exists for 2008-09, it'll always be entirely null for 2020/2021 runs.

Reply:
Ah, yeah, I guess that will be the case, good thought!

I think the fast way around this is to have `cols_that_are_allowed_to_be_all_null` or some better name, and include the annual rate cols. WDYT?

Reply:
Seems fine to me. Other alternative is to only run this check on the full ETL which would lose some information.

Reply:
I think this is going to be a pervasive issue with the data validation style asset checks -- we may have very different expectations about column contents if they contain one year or all the years of data.

Reply:
['annual_average_consumption_rate_gallons_per_minute', 'annual_average_discharge_rate_gallons_per_minute', 'annual_average_withdrawal_rate_gallons_per_minute', 'annual_maximum_intake_summer_temperature_fahrenheit', 'annual_maximum_intake_winter_temperature_fahrenheit', 'annual_maximum_outlet_summer_temperature_fahrenheit', 'annual_maximum_outlet_winter_temperature_fahrenheit', 'annual_total_chlorine_lbs'] 
are all the columns that fail on the fast ETL.

I wonder if we can use job ID or the years of the run to fine-tune which validation tests run for what, or define 'full ETL only' checks like our current validation tests. Possibly @bendnorman has thoughts here. But this is already a 1k line PR and I'm wondering if we just want to scope down here and leave this as a to-do in a separate issue.

Reply:
I bet we can access the dagster `context` object within the asset check and grab the etl configuration or job id. Let me check.

Reply:
Ok! It looks like we can access `context` inside of an `@asset_check` (I think all Dagster concepts are just `@op`'s under the hood). With `context`, we can access the `dataset_settings` resource and the current `job_name`, which we could use to toggle between different sets of checks for a given asset and set of years.

However, IDK if `context.job_name` is the most reliable thing to use for determining what asset checks should be run. I think the name can get lost based on how you execute an asset. For example, while testing this out I executed a single asset in the `etl_fast` job and the `job_name` was "__ASSET_JOB". We're probably better off using the `dataset_settings` to toggle between different sets of checks for an asset.

For assets with metadata, we could probably store multiple sets of expected values given a set of years. I'm not sure how we would do this for assets without metadata though.

@jdangerx do you have a vision for storing/organizing asset check metadata?